File: 169_Majority_Element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (nums) {
    var count = {};
    var m = 0;
    nums.forEach(function (v) {
        if (count[v]) {
            count[v]++;
            if (count[v] > count[m]) {
                m = v;
            }
        } else {
            count[v] = 1;
            if (!count[m]) {
                m = v;
            }
        }
    });

    return m;
};

File: 169_Majority_Element.c
int majorityElement(int *nums, int numsSize) {
  int cnt = 0;
  int val = nums[0];
  for (int i = 0; i < numsSize; i++) {
    if (val == nums[i]) cnt++;
    if (val != nums[i]) {
      if (cnt >= 1) {
        cnt--;
        if (cnt == 0) val = nums[1 + i];
      }
    }
  }

  return val;
}
